### DISCUSSION TITLE
How to make a run-time error when using a Scoped Provider that is not overridden?

### DISCUSSION THREAD

**Discussion by hisaichi5518** [rrousselGit/riverpod#3667] [Q&A] [answered]
I want to create a Scoped Provider that returns a Notifier using code generation. I want this Scoped Provider to make a runtime error if it is not overridden by a ProviderScope, but how should I achieve this?

I have created the following Notifier. The initial value should always be 0.

```dart
@Riverpod(dependencies: [])
class CountNotifier extends _$CountNotifier {
  @override
  int build() {
    return 0;
  }

  void increment() {
    state = state + 1;
  }
}
```

It can be treated as a Scoped Provider by using ProviderScope as follows.

```dart
ProviderScope(
  overrides: [
    countNotifierProvider.overrideWith(CountNotifier.new),
  ],
  child: MyScreen._(keyword: keyword),
)
```
However, when I forget to override it, it works, not a runtime error How can I make it a runtime error when I don't override the Scoped Provider?

**Comment by snapsl** [comment, marked as answer]
You can enforce this by overwriting in the root providerscope as follows

```dart
    ProviderScope(
      overrides: [
        countNotifierProvider.overrideWith(() => throw UnimplementedError()),
      ],
    );
```

**Reply by hisaichi5518** [reply]
I see. In the case of my app, there are many such scoped providers across multiple screens. So I'm starting to think that it might be a good idea to stop code generation and define providers that can only be used by overriding them.

**Reply by snapsl** [reply]
I would not draw this conclusion as code generation will like be the future-proof api.

**Reply by hisaichi5518** [reply]
Yes you are right, I will reconsider including not using scoped provider. Thank you!
